pkg/services/accesscontrol/resourcepermissions/service.go

```diff
@@ -132,16 +137,58 @@ func (s *Service) GetPermissions(ctx context.Context, user identity.Requester, r
 		}
 	}
 
-	return s.store.GetResourcePermissions(ctx, user.GetOrgID(), GetResourcePermissionsQuery{
+	actions := s.actions
+	if s.features.IsEnabled(ctx, featuremgmt.FlagAccessActionSets) {
+		for _, action := range s.actions {
+			actionSets := s.actionSetSvc.ResolveAction(action)
+			for _, actionSet := range actionSets {
+				if !slices.Contains(actions, actionSet) {
+					actions = append(actions, actionSet)
+				}
+
+			}
+		}
+	}
+
+	resourcePermissions, err := s.store.GetResourcePermissions(ctx, user.GetOrgID(), GetResourcePermissionsQuery{
 		User:                 user,
-		Actions:              s.actions,
+		Actions:              actions,
 		Resource:             s.options.Resource,
 		ResourceID:           resourceID,
 		ResourceAttribute:    s.options.ResourceAttribute,
 		InheritedScopes:      inheritedScopes,
 		OnlyManaged:          s.options.OnlyManaged,
 		EnforceAccessControl: s.license.FeatureEnabled("accesscontrol.enforcement"),
 	})
+	if err != nil {
+		return nil, err
+	}
+
+	if s.features.IsEnabled(ctx, featuremgmt.FlagAccessActionSets) {
+		for i := range resourcePermissions {
+			actions := resourcePermissions[i].Actions
+			var expandedActions []string
+			for _, action := range actions {
+				if isFolderOrDashboardAction(action) {
+					actionSetActions := s.actionSetSvc.ResolveActionSet(action)
+					if len(actionSetActions) > 0 {
```

Comment:
I think this is redundant (`for...range` will only iterate if len > 0)
```suggestion
```

Reply:
Ah, sorry, seems the purpose is to break the loop in case of action set is not empty